opam-ci-check/bin/main.ml

```diff
@@ -16,12 +18,68 @@ let to_exit_code : (unit, string) result Term.t -> Cmd.Exit.code Term.t =
       Printf.eprintf "%s%!" msg;
       1
 
+let download_package ~pkg ~url f =
+  Dir_helpers.with_temp_dir "lint-dune-project-version-" @@ fun dir ->
+  let res =
+    OpamProcess.Job.run
+    @@ OpamRepository.pull_tree
+         (OpamPackage.to_string pkg)
+         (OpamFilename.Dir.of_string dir)
+         (OpamFile.URL.checksum url)
+         [ OpamFile.URL.url url ]
+  in
+  let dir =
+    match res with
+    | OpamTypes.Not_available (_, msg) ->
+        print_endline msg;
+        None
+    | Up_to_date _ | Result _ -> Some dir
+  in
+  f dir
+
+let read_package_opam ~opam_repo_dir pkg =
+  let opam_path =
+    Opam_helpers.path_from_pkg ~repo_dir:opam_repo_dir pkg // "opam"
+  in
+  (* NOTE: We use OpamFile.OPAM.read_from_channel instead of OpamFile.OPAM.file
+     to prevent the name and version fields being automatically added *)
+  In_channel.with_open_text opam_path (fun ic ->
+      let opam =
+        try Ok (OpamFile.OPAM.read_from_channel ic)
+        with OpamPp.Bad_format e | OpamPp.Bad_version (e, _) -> Error e
+      in
+      opam)
```

Comment:
I find the `opam` variable a bit misleading here, because it may actually be an `Error _`, and so not have opam file data at all. IMO this would be clearer like

```suggestion
        try Ok (OpamFile.OPAM.read_from_channel ic)
        with OpamPp.Bad_format e | OpamPp.Bad_version (e, _) -> Error e)
```

or maybe

```suggestion
        try     
          let opam = OpamFile.OPAM.read_from_channel ic in
          Ok opam
        with 
          OpamPp.Bad_format e | OpamPp.Bad_version (e, _) -> Error e)
```

or maybe rename the `opam` variable to `result`?